src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,69 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.impl.ImplementationFinder;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private final Map<Class<?>, Object> instances = new HashMap<>();
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        if (interfaceClazz.isAnnotationPresent(Component.class)) {
+            Object clazzImplementationInstance = null;
+            Class<?> clazz = findImplementation(interfaceClazz);
+            Field[] declaredFields = clazz.getDeclaredFields();
+            for (Field field : declaredFields) {
+                if (field.isAnnotationPresent(Inject.class)) {
+                    Object fieldInstance = getInstance(field.getType());
+                    clazzImplementationInstance = createNewInstance(clazz);
+                    field.setAccessible(true);
+                    try {
+                        field.set(clazzImplementationInstance, fieldInstance);
+                    } catch (IllegalAccessException e) {
+                        throw new RuntimeException("Can't initialize field value. Class: "
+                                + clazz.getName() + ". Field: " + field.getName(), e);
+                    }
+                }
+            }
```

Comment:
The logic within the for-loop for setting field values should be after the check for `clazzImplementationInstance` being `null`. Otherwise, you may attempt to set fields on a `null` reference if `createNewInstance` isn't called first.

Reply:
But if I move the checking if clazzImplementationInstance is null, then the condition will always be true..
Ok, anyway I moved it...